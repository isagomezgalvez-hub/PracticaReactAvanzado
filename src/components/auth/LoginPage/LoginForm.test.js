import React from 'react';
import { shallow } from 'enzyme';

import LoginForm from './LoginForm';


describe('LoginForm', ()=>{
	const props = {
		isLoading: false,
		onSubmit: jest.fn(),
	}
	const render = () => shallow(<LoginForm {...props} />);
	test('should render',()=>{
		const wrapper = render()
		expect(wrapper.exists()).toBe(true)
	})
	test('snapshot testing of LoginForm', ()=>{
		const wrapper= render()
		expect(wrapper).toMatchSnapshot();
	})

	test('should submit credentials - test acción del store', ()=>{
		const credentials = { email: 'isa', password: 'password', remember:false}
		const wrapper = render();

		const usernameField = wrapper.find('.loginForm-field').at(0);
		usernameField.props().onChange({target:{name: 'email', value: credentials.email}})

		const passwordField = wrapper.find('.loginForm-field').at(0);
		usernameField.props().onChange({ target: {name:'password', value: credentials.password }  })

		const form = wrapper.find('form');
		form.simulate('submit', {preventDefault: ()=> {}});

		expect(wrapper.find('.loginForm-submit').props().disabled).toBe(false)
		expect(props.onSubmit).toHaveBeenCalledWith(credentials);
		
	} )

	
})