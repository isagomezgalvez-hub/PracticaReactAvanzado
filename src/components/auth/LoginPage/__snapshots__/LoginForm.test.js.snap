// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm snapshot testing of LoginForm 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="email"
    onChange={[Function]}
    value=""
  />
  <input
    className="loginForm-field"
    name="password"
    onChange={[Function]}
    type="password"
    value=""
  />
  <input
    checked={false}
    className="loginForm-field"
    name="remember"
    onChange={[Function]}
    type="checkbox"
  />
  <button
    className="loginForm-submit"
    disabled={true}
  >
    Login
  </button>
</form>
`;
